backend/routes/seatbid.js

```diff
@@ -0,0 +1,186 @@
+
+const prisma = new PrismaClient();
+const router = express.Router();
+
+/**
+ * @route POST /start
+ * @desc Start a new bidding for a seat
+ * @access Private
+ */
+router.post('/start', authenticateUser, async (req, res) => {
+    const { passengerId, seatNumber, flightId } = req.body;
+    const biddingId = `${flightId}-${seatNumber}`;
+    const expirationTime = Date.now() + 30 * 60 * 1000; // 30 minutes
+
+    const existingBidding = Array.from(activeBiddings.values()).find(
+        bidding => bidding.passengerId === passengerId && bidding.status === 'ACTIVE'
+    );
+
+    if (existingBidding) {
+        return res.status(400).json({ error: 'You already have an active bidding' });
+    }
+
+    const newBidding = {
+        passengerId,
+        seatNumber,
+        flightId,
+        startTime: Date.now(),
+        expirationTime,
+        status: 'ACTIVE'
+    };
+
+    activeBiddings.set(biddingId, newBidding);
+    bids.set(biddingId, []);
+
+    emitToFlight(req.app.locals.io, flightId, 'new-bid', { 
+        biddingId, 
+        seatNumber, 
+        passengerId,
+        bids: []
+    });
+
+    setTimeout(() => expireBidding(req.app.locals.io, biddingId), expirationTime - Date.now());
+
+    res.json({ success: true, biddingId, ...newBidding, bids: [] });
+});
+
+/**
+ * @route POST /bid
+ * @desc Place a bid on an active bidding
+ * @access Private
+ */
+router.post('/bid', authenticateUser, async (req, res) => {
+    const { biddingId, bidderId, amount, bidderSeatNumber } = req.body;
+    const bidding = activeBiddings.get(biddingId);
+
+    if (!bidding || bidding.status !== 'ACTIVE') {
+        return res.status(400).json({ error: 'Bidding not found or not active' });
+    }
+
+    const bidderActiveBidding = Array.from(activeBiddings.values()).find(
+        b => b.passengerId === bidderId && b.status === 'ACTIVE'
+    );
+
+    if (bidderActiveBidding) {
+        bidderActiveBidding.status = 'CANCELLED';
+        emitToFlight(req.app.locals.io, bidderActiveBidding.flightId, 'bidding-cancelled', { biddingId: bidderActiveBidding.biddingId });
+    }
+
+    const bid = { 
+        id: Date.now().toString(), 
+        bidderId, 
+        amount, 
+        bidderSeatNumber, 
+        time: Date.now() 
+    };
+
+    if (!bids.has(biddingId)) bids.set(biddingId, []);
+    bids.get(biddingId).push(bid);
+
+    emitToFlight(req.app.locals.io, bidding.flightId, 'new-bid', { biddingId, bid });
+
+    res.json({ success: true, bidId: bid.id });
+});
+
+/**
+ * @route POST /accept
+ * @desc Accept a bid and complete the seat swap
+ * @access Private
+ */
+router.post('/accept', authenticateUser, async (req, res) => {
+    const { biddingId, bidId } = req.body;
+    const bidding = activeBiddings.get(biddingId);
+
+    if (!bidding || bidding.status !== 'ACTIVE') {
+        return res.status(400).json({ error: 'Bidding not found or not active' });
+    }
+
+    const acceptedBid = bids.get(biddingId).find(bid => bid.id === bidId);
```

Comment:
Can we also verify the bid owner still owns their seat?

Reply:
added in the next PR